slither/core/expressions/literal.py

```diff
@@ -68,8 +68,3 @@ def __str__(self) -> str:
 
         # be sure to handle any character
         return str(self._value)
-
-    def __eq__(self, other: Any) -> bool:
```

Comment:
As discussed, if we are sure this equality should never be used, we might want to add `assert False` here. It would be a bug if the default `__eq__` is used.

Reply:
Reading the original description, I guess having __eq__ with assertion wouldn't be feasible.